### PR TITLE
Dependency update: update polar to 0.25

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Update polar version to 0.25

# Tests + Formatting

Make sure you've done the following, if applicable:

- Add tests that cover your changes (either in the command examples, the crate/tests folder, or in the /tests folder)
  - Try to think about corner cases and various ways how your changes could break. Cover those in the tests

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace --features=extra` to check that all tests pass

# User-Facing Changes

If you're making changes that will affect the user experience of Nushell (ex: adding/removing a command, changing an input/output type, adding a new flag):

- Get another regular contributor to review the PR before merging
- Make sure that there is an entry in the documentation (https://github.com/nushell/nushell.github.io) for the feature, and update it if necessary

### CODE REVIEW

File: crates/nu-command/src/dataframe/values/nu_expression/mod.rs [hunk 1]
```diff
@@ -174,16 +174,6 @@ pub fn expr_to_value(expr: &Expr, span: Span) -> Value {
     let cols = vec!["expr".to_string(), "value".to_string()];
 
     match expr {
-        Expr::Not(expr) => {
```
Comment (fdncred) [inline crates/nu-command/src/dataframe/values/nu_expression/mod.rs:177]: It seems like we're losing several expressions throughout this file. Is that because they don't exist anymore in polars or some other reason? I also wonder if there's a different way to do this now, since I'm assuming they just don't exist anymore in the Expr enum.

**Comment by stormasm** [comment]
@elferherrera I know you are super busy, but if you have a bit of time it would be great if you could quickly review this PR.

If not, no problem, I will understand...

Thanks !

File: crates/nu-command/src/dataframe/values/nu_expression/mod.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/dataframe/values/nu_expression/mod.rs:177]: Polar moves these expr variant from `Expr` into [FunctionExpr](https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html).

And I just take a look about these variant, it seems that we actually have no way to construct these removed `Expr`.

Comment (WindSoilder) [reply crates/nu-command/src/dataframe/values/nu_expression/mod.rs:177]: And after take a closer look about NuExpression's constructors(from nushell value): try_from_value.

It seems that Only specific variants' of polar expression can be constructed

Comment (WindSoilder) [reply crates/nu-command/src/dataframe/values/nu_expression/mod.rs:177]: Ah! That's a good catch!  I need to take a closer look around NuExpression with your question.

Comment (WindSoilder) [reply crates/nu-command/src/dataframe/values/nu_expression/mod.rs:177]: Sorry for waiting for a long time, here is my investigation result:

For the removing, we need to know how `Expr` is constructed from nushell's Value or pipeline input.  

And it's constructed by `NuExpression::try_from_value` and `NuExpression::try_from_pipeline`(which finally leads to `NuExpression::try_from_value`).  In general, nushell takes user argument input, and convert it to `NuExpression` through `try_from_value` method.

After looking into `try_from_value`, it seems that it only accepts `String, Int, Bool, Float` as input.  So no `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` is used.  So the reason they exists is satify our compilder, and don't need to use something like `expect`(to expect these expression never occurred), and we don't want to throws out error.

Underhood, `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` are removed from `Expr` to [FunctionExpr]`https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html`, and I've checked `FunctionExpr` usage in nushel, I think we don't use them in source code.  So it's ok to just take and go through with it.

Comment (WindSoilder) [reply crates/nu-command/src/dataframe/values/nu_expression/mod.rs:177]: For the removing, we need to know how `Expr` is constructed from nushell's Value or pipeline input.  

And it's constructed by `NuExpression::try_from_value` and `NuExpression::try_from_pipeline`(which finally leads to `NuExpression::try_from_value`).  In general, nushell takes user argument input, and convert it to `NuExpression` through `try_from_value` method.

After looking into `try_from_value`, it seems that it only accepts `String, Int, Bool, Float` as input.  So no `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` is used.  So the reason they exists is satify our compilder, and don't need to use something like `expect`(to expect these expression never occurred), and we don't want to throws out error.

Underhood, `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` are removed from `Expr` to [FunctionExpr](https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html), and I've checked `FunctionExpr` usage in nushel, I think we don't use them in source code.  So it's ok to just take and go through with it.

Comment (WindSoilder) [reply crates/nu-command/src/dataframe/values/nu_expression/mod.rs:177]: Sorry for a relative long time, here is my investigation result

For the removing, we need to know how `Expr` is constructed from nushell's Value or pipeline input.  

And it's constructed by `NuExpression::try_from_value` and `NuExpression::try_from_pipeline`(which finally leads to `NuExpression::try_from_value`).  In general, nushell takes user argument input, and convert it to `NuExpression` through `try_from_value` method.

After looking into `try_from_value`, it seems that it only accepts `String, Int, Bool, Float` as input.  So no `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` is used.  So the reason they exists is satify our compilder, and don't need to use something like `expect`(to expect these expression never occurred), and we don't want to throws out error.

Underhood, `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` are removed from `Expr` to [FunctionExpr](https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html), and I've checked `FunctionExpr` usage in nushel, I think we don't use them in source code.  So it's ok to just take and go through with it.

> Is that because they don't exist anymore in polars or some other reason?

So as conclusion: I think they exists just for satifying our compiler, and current version of nushell never execute it
